packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/dataservice/WorkflowController.java

```diff
@@ -223,13 +269,37 @@ public ResponseEntity<ResponseDeleted> deleteWorkflow(
 			projectId
 		);
 
+		final ClientEvent<Void> event = ClientEvent.<Void>builder().type(ClientEventType.WORKFLOW_UPDATE).build();
+
+		final RebacProject rebacProject = new RebacProject(projectId, reBACService);
+
 		try {
 			workflowService.deleteAsset(id, projectId, permission);
-			return ResponseEntity.ok(new ResponseDeleted("Workflow", id));
 		} catch (final Exception e) {
 			final String error = String.format("Failed to delete workflow %s", id);
 			log.error(error, e);
-			throw new ResponseStatusException(org.springframework.http.HttpStatus.INTERNAL_SERVER_ERROR, error);
+			throw new ResponseStatusException(
+				org.springframework.http.HttpStatus.INTERNAL_SERVER_ERROR,
+				messages.get("postgres.service-unavailable")
+			);
 		}
+
+		try {
+			if (rebacProject.isPublic()) {
+				clientEventService.sendToAllUsers(event);
+			} else {
+				final List<String> userIds = projectPermissionsService
+					.getReaders(rebacProject)
+					.stream()
+					.map(Contributor::getUserId)
+					.toList();
+				clientEventService.sendToUsers(event, userIds);
+			}
+		} catch (final Exception e) {
+			log.error("Unable to notify users of deleted  workflow", e);
+			// No response status exception here because the workflow was deleted successfully, and it's just the update that's failed.
+		}
```

Comment:
Why not create a method at the top of this controller that accepts an event and a projectId as arguments? This way, any new endpoints can easily access this logic. Alternatively, I would suggest placing this method in the ClientEvent class, as it simply determines who should read the event based on the projectId.

Reply:
I'm going to tuck that first idea away as the next step if we end up needing more collaboration updates :)